### DISCUSSION TITLE
bug: validator warns to change the `last` in the schedule field value to `first`

### DISCUSSION THREAD

**Discussion by Mogyuchi** [renovatebot/renovate#28995] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

None

### Please tell us more about your question or problem

renovate validator tells us to change `last` to `first` when it validates the below config, which is incorrect.
```json
{
  "schedule": ["on the last day of the month"]
}
```
```console
$ npx --yes --package renovate@37.354.6 -- renovate-config-validator --strict
 INFO: Validating renovate.json
 WARN: Config migration necessary
       "oldConfig": {"schedule": ["on the last day of the month"]},
       "newConfig": {"schedule": ["on the first day of the month"]}
```

**Comment by rarkins** [comment]
It's an intentional migration, but I don't recall why. Maybe the underlying library we have had a problem with it. When you pick first day, it's always 1. But last day could be 28,29,30,31.

**Reply by HonkingGoose** [reply]
> It's an intentional migration, but I don't recall why.

You created this PR:

- #680

To fix issue:

- #1047

In that issue you said: [^1]

> Thank you for reporting the bug. The logs from this repo have helped me find that our existing "is this branch scheduled now?" algorithm *ignores* day of the month and is only checking day of the week and time of day. So your schedule basically becomes "before 3am". I'll fix!

[^1]: [Source for quote](https://github.com/renovatebot/renovate/issues/1047#issuecomment-339268015)

**Reply by Mogyuchi** [reply]
@rarkins 
It should also work `on the last day of the month` since it has been changed to use the built-in `isValid` at https://github.com/renovatebot/renovate/pull/9819 .
For information, later uses `isValid` to test `on the last day of the month`.
https://github.com/breejs/later/blob/v4.2.0/test/example/text-example-test.js#L339-L363

**Reply by rarkins** [reply]
You're welcome to submit a PR if you can test that it works as expected
